shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm

```diff
@@ -70,6 +76,77 @@ - (BOOL)pointInside:(CGPoint)point withEvent:(UIEvent*)event {
   return NO;
 }
 
+// Creates and initializes a UIVisualEffectView with a UIBlurEffect. Extracts and returns its
+// gaussianFilter. Logs errors and returns if Apple's API has changed and the filter can't be
+// extracted.
+- (NSObject*)extractGaussianFilter {
+  UIVisualEffectView* visualEffectView = [[UIVisualEffectView alloc]
+      initWithEffect:[UIBlurEffect effectWithStyle:UIBlurEffectStyleLight]];
+
+  NSObject* gaussianFilter =
+      [[[visualEffectView.subviews firstObject].layer.filters firstObject] retain];
+  if (!gaussianFilter || ![[gaussianFilter valueForKey:@"name"] isEqual:@"gaussianBlur"]) {
+    FML_DLOG(ERROR) << "Apple's API for UIVisualEffectView changed. Update the implementation to "
+                       "access the Gaussian blur filter. ";
+    return nil;
+  }
+
+  [visualEffectView release];
+
+  if (![[gaussianFilter valueForKey:@"inputRadius"] isKindOfClass:[NSNumber class]]) {
+    FML_DLOG(ERROR) << "Apple's API for UIVisualEffectView changed. Update the implementation "
+                       "access the Gaussian blur filter's properties.";
+    return nil;
+  }
+
+  return gaussianFilter;
+}
+
+- (void)applyBackdropFilters:(NSArray*)blurRadii {
+  if (!_activeGaussianFilters) {
+    _activeGaussianFilters = [[[NSMutableArray alloc] init] retain];
+
+    _gaussianFilter = [self extractGaussianFilter];
+    if (!_gaussianFilter) {
+      return;
+    }
+  }
+
+  bool updatedFilters = false;
+
+  // Update the size of _activeGaussianFilters to match the number of applied backdrop filters.
+  while ([blurRadii count] > [_activeGaussianFilters count]) {
+    // copy returns a deep copy of _gaussianFilter
+    [_activeGaussianFilters addObject:[_gaussianFilter copy]];
+    updatedFilters = true;
+  }
+  while ([blurRadii count] < [_activeGaussianFilters count]) {
+    [_activeGaussianFilters removeLastObject];
+    updatedFilters = true;
+  }
+
+  for (NSUInteger i = 0; i < [blurRadii count]; i++) {
+    if ([_activeGaussianFilters[i] valueForKey:@"inputRadius"] == blurRadii[i]) {
+      continue;
+    }
+    [_activeGaussianFilters[i] setValue:blurRadii[i] forKey:@"inputRadius"];
+    updatedFilters = true;
+  }
+
+  if (updatedFilters) {
+    self.layer.filters = _activeGaussianFilters;
```

Comment:
Could you make sure this properly triggers the `-[CALayer setFilter:]` setter and applies the new filters? Because we're reusing the same `NSArray` instance and the setter may have logic like this:
```objc
- (void)setFilters: (NSArray*)newFilters { 
  if (_filters == newFIlters) { 
    return;
  }
  // update filters for real.
}
```

Reply:
Also according to the discussion section in the [documentation](https://developer.apple.com/documentation/quartzcore/calayer/1410901-filters?language=objc), you'll have to set the filters in a differnet way in order to update them. Could you verify that the current implementation works when the user changes the input sigma?

Reply:
We wrote the visual test and confirmed your suspicion - the setter was not being called as we expected. We updated the code to make copies of the _gaussianBlur filter and fill _activeGaussianFilters with those copies. This change is working without making copies of _activeGaussianFilters since the original array is updated.